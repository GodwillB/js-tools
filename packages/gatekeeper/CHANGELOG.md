# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## [0.0.3](https://github.com/onaio/js-tools/compare/@onaio/gatekeeper@0.0.2...@onaio/gatekeeper@0.0.3) (2019-04-26)

**Note:** Version bump only for package @onaio/gatekeeper

## [0.0.2](https://github.com/onaio/js-tools/compare/@onaio/gatekeeper@0.0.1...@onaio/gatekeeper@0.0.2) (2019-04-24)

**Note:** Version bump only for package @onaio/gatekeeper

## 0.0.1 (2019-04-23)

**Note:** Version bump only for package @onaio/gatekeeper
