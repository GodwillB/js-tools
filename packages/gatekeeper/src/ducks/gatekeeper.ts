import { ActionCreator, AnyAction, Store } from 'redux';
import SeamlessImmutable from 'seamless-immutable';

export const reducerName = 'gatekeeper';

/** Interface for Record action */
export interface RecordAction extends AnyAction {
  result: { [key: string]: any };
  success: boolean | null;
  type: typeof RECORD;
}

/** interface to describe GateKeeper state */
export interface GateKeeperState {
  result: { [key: string]: any } /** stores the result of the auth attempt */;
  success: boolean | null /** was it successful or not */;
}

/** immutable GateKeeper state */
export type ImmutableGateKeeperState = GateKeeperState &
  SeamlessImmutable.ImmutableObject<GateKeeperState>;

/** Initial state for GateKeeper */
export const initialState: ImmutableGateKeeperState = SeamlessImmutable({
  result: {},
  success: null
});

/** GateKeeper reducer function
 * @param {initialState} state - the initial state
 * @param {AnyAction} action - the action
 */
export default function reducer(state = initialState, action: AnyAction): ImmutableGateKeeperState {
  switch (action.type) {
    case RECORD:
      return state.merge({
        result: { ...action.result },
        success: action.success
      });
    default:
      return state;
  }
}

// actions

/** authenticate success action type */
export const RECORD = '@onaio/gatekeeper/reducer/RECORD';

// action creators
/** record the result of the authentication attempt
 * @param {boolean} success - whether it was successful or not
 * @param {{ [key: string]: any }} result - an object containing result information
 */
export const recordResult: ActionCreator<RecordAction> = (
  success: boolean,
  result: { [key: string]: any } = {}
) => ({
  result,
  success,
  type: RECORD
});

// selectors

/** get result
 * @param {Partial<Store>} state - the redux store
 */
export function getResult(state: Partial<Store>): { [key: string]: any } {
  return (state as any)[reducerName].result;
}

/** get success
 * @param {Partial<Store>} state - the redux store
 */
export function getSuccess(state: Partial<Store>): boolean {
  return (state as any)[reducerName].success;
}
