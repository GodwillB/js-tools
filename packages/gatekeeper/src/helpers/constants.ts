export const ONADATA_PROVIDER_NAME = 'onadata';
export type ONADATA_PROVIDER_NAME = typeof ONADATA_PROVIDER_NAME;
