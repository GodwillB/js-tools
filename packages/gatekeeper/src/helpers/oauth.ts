import ClientOAuth2 from 'client-oauth2';

export const onadataAuth = new ClientOAuth2({
  accessTokenUri: 'https://stage-api.ona.io/o/token/',
  authorizationUri: 'https://stage-api.ona.io/o/authorize/',
  clientId: 'iDWxqlTdUNUtGd3MIjxsI8HcICSjsRwHfy1LKvzF',
  redirectUri: 'http://localhost:3000/oauth/callback/onadata/',
  scopes: ['read', 'write'],
  state: 'abc'
});
