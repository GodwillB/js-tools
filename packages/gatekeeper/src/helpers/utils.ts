/** Type definition for error callback function  */
export type ErrorCallback = (e: string) => string | void;

/** Error callback function
 * @param {string} error - the error message
 */
export function errorCallback(error: string) {
  throw new Error(error);
}
