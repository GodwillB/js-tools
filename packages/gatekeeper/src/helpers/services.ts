import { AuthenticateAction, authenticateUser } from '@onaio/session-reducer';
import ClientOAuth2 from 'client-oauth2';
import { Dispatch } from 'react';
import { ActionCreator } from 'redux';
import { getOnadataUserInfo, UserInfoFnType } from './oauth';
import { ErrorCallback, errorCallback } from './utils';

/** Calls the oAuth provider to get user details
 * @param {string} locationHash - the location hash value that we receive from the oAuth provider
 * @param {string} url - the URL that returns the user information for the provider
 * @param {ClientOAuth2} provider - the Oauth client object for the provider
 * @param {UserInfoFnType} userInfoCallback - function the gets user info from API response
 */
export async function oauth2Callback(
  locationHash: string,
  url: string,
  provider: ClientOAuth2,
  userInfoCallback: UserInfoFnType
) {
  return provider.token.getToken(locationHash).then(async (oAuthObject: any) => {
    const response = await fetch(
      url,
      oAuthObject.sign({
        method: 'GET',
        url
      })
    );

    if (!response.ok) {
      throw new Error(`oAuth service oauth2Callback failed, HTTP status ${response.status}`);
    }

    const data = await response.json();

    return userInfoCallback(data);
  });
}

/** This function is used to fetch the user logging in by calling oauth2Callback
 * and then calling authenticateUser to store the user in the Redux store.
 * @param {string} locationHash - the location hash value that we receive from the oAuth provider
 * @param {string} url - the URL that returns the user information for the provider
 * @param {any} dispatch - the dispatch function that calls the session reducer action creator
 * @param {ClientOAuth2} provider - the Oauth client object for the provider
 * @param {UserInfoFnType} userInfoCallback - function the gets user info from API response
 * @param {ErrorCallback} errorCallbackFn - a function that handles error messages
 */
export async function fetchUser(
  locationHash: string,
  url: string,
  dispatch: Dispatch<any>,
  provider: ClientOAuth2,
  authenticateActionCreator: ActionCreator<AuthenticateAction> = authenticateUser,
  userInfoCallback: UserInfoFnType = getOnadataUserInfo,
  errorCallbackFn: ErrorCallback = errorCallback
) {
  try {
    const userInfo = await oauth2Callback(locationHash, url, provider, userInfoCallback);
    if (userInfo) {
      const { authenticated, user, extraData } = userInfo;
      dispatch(authenticateActionCreator(authenticated, user, extraData));
    } else {
      errorCallbackFn('Something went wrong');
    }
  } catch (error) {
    errorCallbackFn(error.message);
  }
}
