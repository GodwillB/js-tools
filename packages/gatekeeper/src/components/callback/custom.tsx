import {
  AuthenticateAction,
  authenticateUser,
  getExtraData,
  getUser,
  isAuthenticated
} from '@onaio/session-reducer';
import React, { useEffect } from 'react';
import { ActionCreator, Store } from 'redux';
import { getSuccess, RecordAction, recordResult } from '../../ducks/gatekeeper';
import { fetchState } from '../../helpers/services';
import { errorCallback } from '../../helpers/utils';
import {
  BaseCallbackComponent,
  BaseCallbackComponentProps,
  defaultBaseCallbackComponentProps
} from './base';
import { Component404, RenderErrorComponent } from './utils';

interface RouteParams {
  id: string;
}

/** interface for OauthCallbackProps props */
export interface OauthCallbackProps<G> extends BaseCallbackComponentProps<G> {
  ErrorComponent: React.ElementType;
  HTTP404Component: React.ElementType;
  apiURL: string;
  authenticateActionCreator: ActionCreator<AuthenticateAction>;
  recordResultActionCreator: ActionCreator<RecordAction>;
}

/** default props for OauthCallback */
export const defaultOauthCallbackProps: Partial<OauthCallbackProps<RouteParams>> = {
  ...defaultBaseCallbackComponentProps,
  ErrorComponent: RenderErrorComponent,
  HTTP404Component: Component404,
  UnSuccessfulLoginComponent: RenderErrorComponent,
  authenticateActionCreator: authenticateUser,
  recordResultActionCreator: recordResult
};

const CustomCallback = (props: OauthCallbackProps<RouteParams>) => {
  const {
    LoadingComponent,
    SuccessfulLoginComponent,
    UnSuccessfulLoginComponent,
    apiURL,
    authSuccess,
    authenticated,
    authenticateActionCreator,
    recordResultActionCreator,
    sessionData,
    sessionUser
  } = props;

  useEffect(() => {
    if (authSuccess === null || authenticated === false) {
      fetchState(apiURL, authenticateActionCreator, recordResultActionCreator, errorCallback).catch(
        e => {
          /** do nothing - is this wise?? */
        }
      );
    }
  }, []); // The empty array causes this effect to only run on mount

  const baseProps = {
    LoadingComponent,
    SuccessfulLoginComponent,
    UnSuccessfulLoginComponent,
    authSuccess,
    authenticated,
    sessionData,
    sessionUser
  };

  return <BaseCallbackComponent {...baseProps} />;
};

CustomCallback.defaultProps = defaultOauthCallbackProps;

export { CustomCallback };
