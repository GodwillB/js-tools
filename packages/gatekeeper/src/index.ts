import OauthLogin, { OauthLoginProps, ProviderLinks, ProviderLinksProps } from './components/login';

import ConnectedOauthCallback, {
  Component404,
  OauthCallback,
  OauthCallbackProps,
  RenderErrorComponent,
  RouteParams,
  SuccessfulLogin,
  SuccessfulLoginProps
} from './components/callback';

import {
  getOnadataUserInfo,
  getProviderFromOptions,
  OauthOptions,
  Providers,
  UserInfoFnType
} from './helpers/oauth';

import { fetchUser, oauth2Callback } from './helpers/services';

import { errorCallback, ErrorCallback } from './helpers/utils';

export {
  errorCallback,
  fetchUser,
  getOnadataUserInfo,
  getProviderFromOptions,
  oauth2Callback,
  Component404,
  ConnectedOauthCallback,
  ErrorCallback,
  OauthCallback,
  OauthCallbackProps,
  OauthLogin,
  OauthLoginProps,
  OauthOptions,
  Providers,
  ProviderLinks,
  ProviderLinksProps,
  SuccessfulLogin,
  SuccessfulLoginProps,
  RenderErrorComponent,
  RouteParams,
  UserInfoFnType
};
