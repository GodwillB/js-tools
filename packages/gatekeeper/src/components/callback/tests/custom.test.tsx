import store from '@onaio/connected-reducer-registry';
import reducerRegistry from '@onaio/redux-reducer-registry';
import session, {
  authenticateUser,
  logOutUser,
  reducerName as sessionReducer
} from '@onaio/session-reducer';
import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import fetchMock from 'fetch-mock';
import { createBrowserHistory } from 'history';
import React from 'react';
import { Provider } from 'react-redux';
import gatekeeper, {
  recordResult,
  reducerName as gateKeeperReducer
} from '../../../ducks/gatekeeper';
import { getOnadataUserInfo } from '../../../helpers/oauth';
import * as serviceHelpers from '../../../helpers/services';
import * as helperFixtures from '../../../helpers/tests/fixtures';
import * as fixtures from '../../tests/fixtures';
import * as callback from '../custom';

const CustomCallback = callback.CustomCallback;

const history = createBrowserHistory();
reducerRegistry.register(sessionReducer, session);
reducerRegistry.register(gateKeeperReducer, gatekeeper);

describe('gatekeeper/custom/CustomCallback', () => {
  beforeEach(() => {
    jest.resetAllMocks();
    fetchMock.restore();
  });

  it('renders without crashing', () => {
    fetchMock.getOnce('http://example.com', JSON.stringify(helperFixtures.expressAPIResponse));
    const props = {
      apiURL: 'http://example.com',
      authSuccess: true,
      authenticateActionCreator: authenticateUser,
      authenticated: false,
      sessionData: {},
      sessionUser: {
        email: '',
        gravatar: '',
        name: '',
        username: ''
      }
    };
    shallow(<CustomCallback {...props} />);
  });

  it('renders correctly', () => {
    fetchMock.getOnce('http://example.com', JSON.stringify(helperFixtures.expressAPIResponse));
    const props = {
      apiURL: 'http://example.com',
      authSuccess: true,
      authenticateActionCreator: authenticateUser,
      authenticated: true,
      sessionData: helperFixtures.onadataSession,
      sessionUser: helperFixtures.onadataUser
    };
    store.dispatch(logOutUser());
    const wrapper = mount(<CustomCallback {...props} />);
    expect(toJson(wrapper.find('CustomCallback'))).toMatchSnapshot();
    wrapper.unmount();
  });
});
