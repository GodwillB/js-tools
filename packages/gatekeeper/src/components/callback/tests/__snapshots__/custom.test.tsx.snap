// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gatekeeper/custom/APICallback renders correctly 1`] = `
<APICallback
  ErrorComponent={[Function]}
  HTTP404Component={[Function]}
  LoadingComponent={[Function]}
  SuccessfulLoginComponent={[Function]}
  UnSuccessfulLoginComponent={[Function]}
  apiURL="http://example.com"
  authSuccess={true}
  authenticateActionCreator={[Function]}
  authenticated={true}
  recordResultActionCreator={[Function]}
  sessionData={
    Object {
      "authenticated": true,
      "extraData": Object {
        "api_token": "the api token",
        "city": "",
        "country": "",
        "email": "mosh@example.com",
        "gravatar": "https://secure.gravatar.com/avatar/ae22ab897231db07205bd5d00e64cbbf?d=https%3A%2F%2Fona.io%2Fstatic%2Fimages%2Fdefault_avatar.png&s=60",
        "name": "mosh",
        "oAuth2Data": Object {
          "access_token": "iLoveOov",
          "expires_in": "36000",
          "scope": "read write",
          "state": "abc",
          "token_type": "Bearer",
        },
        "organization": "",
        "require_auth": true,
        "temp_token": "the temp token",
        "twitter": "",
        "url": "https://stage-api.ona.io/api/v1/profiles/moshthepitt",
        "user": "https://stage-api.ona.io/api/v1/users/moshthepitt",
        "username": "moshthepitt",
        "website": "",
      },
      "user": Object {
        "email": "mosh@example.com",
        "gravatar": "https://secure.gravatar.com/avatar/ae22ab897231db07205bd5d00e64cbbf?d=https%3A%2F%2Fona.io%2Fstatic%2Fimages%2Fdefault_avatar.png&s=60",
        "name": "mosh",
        "username": "moshthepitt",
      },
    }
  }
  sessionUser={
    Object {
      "api_token": "the api token",
      "city": "",
      "country": "",
      "email": "mosh@example.com",
      "gravatar": "https://secure.gravatar.com/avatar/ae22ab897231db07205bd5d00e64cbbf?d=https%3A%2F%2Fona.io%2Fstatic%2Fimages%2Fdefault_avatar.png&s=60",
      "name": "mosh",
      "organization": "",
      "require_auth": true,
      "temp_token": "the temp token",
      "twitter": "",
      "url": "https://stage-api.ona.io/api/v1/profiles/moshthepitt",
      "user": "https://stage-api.ona.io/api/v1/users/moshthepitt",
      "username": "moshthepitt",
      "website": "",
    }
  }
>
  <BaseCallbackComponent
    LoadingComponent={[Function]}
    SuccessfulLoginComponent={[Function]}
    UnSuccessfulLoginComponent={[Function]}
    authSuccess={true}
    authenticated={true}
    sessionData={
      Object {
        "authenticated": true,
        "extraData": Object {
          "api_token": "the api token",
          "city": "",
          "country": "",
          "email": "mosh@example.com",
          "gravatar": "https://secure.gravatar.com/avatar/ae22ab897231db07205bd5d00e64cbbf?d=https%3A%2F%2Fona.io%2Fstatic%2Fimages%2Fdefault_avatar.png&s=60",
          "name": "mosh",
          "oAuth2Data": Object {
            "access_token": "iLoveOov",
            "expires_in": "36000",
            "scope": "read write",
            "state": "abc",
            "token_type": "Bearer",
          },
          "organization": "",
          "require_auth": true,
          "temp_token": "the temp token",
          "twitter": "",
          "url": "https://stage-api.ona.io/api/v1/profiles/moshthepitt",
          "user": "https://stage-api.ona.io/api/v1/users/moshthepitt",
          "username": "moshthepitt",
          "website": "",
        },
        "user": Object {
          "email": "mosh@example.com",
          "gravatar": "https://secure.gravatar.com/avatar/ae22ab897231db07205bd5d00e64cbbf?d=https%3A%2F%2Fona.io%2Fstatic%2Fimages%2Fdefault_avatar.png&s=60",
          "name": "mosh",
          "username": "moshthepitt",
        },
      }
    }
    sessionUser={
      Object {
        "api_token": "the api token",
        "city": "",
        "country": "",
        "email": "mosh@example.com",
        "gravatar": "https://secure.gravatar.com/avatar/ae22ab897231db07205bd5d00e64cbbf?d=https%3A%2F%2Fona.io%2Fstatic%2Fimages%2Fdefault_avatar.png&s=60",
        "name": "mosh",
        "organization": "",
        "require_auth": true,
        "temp_token": "the temp token",
        "twitter": "",
        "url": "https://stage-api.ona.io/api/v1/profiles/moshthepitt",
        "user": "https://stage-api.ona.io/api/v1/users/moshthepitt",
        "username": "moshthepitt",
        "website": "",
      }
    }
  >
    <SuccessfulLogin
      extraData={
        Object {
          "authenticated": true,
          "extraData": Object {
            "api_token": "the api token",
            "city": "",
            "country": "",
            "email": "mosh@example.com",
            "gravatar": "https://secure.gravatar.com/avatar/ae22ab897231db07205bd5d00e64cbbf?d=https%3A%2F%2Fona.io%2Fstatic%2Fimages%2Fdefault_avatar.png&s=60",
            "name": "mosh",
            "oAuth2Data": Object {
              "access_token": "iLoveOov",
              "expires_in": "36000",
              "scope": "read write",
              "state": "abc",
              "token_type": "Bearer",
            },
            "organization": "",
            "require_auth": true,
            "temp_token": "the temp token",
            "twitter": "",
            "url": "https://stage-api.ona.io/api/v1/profiles/moshthepitt",
            "user": "https://stage-api.ona.io/api/v1/users/moshthepitt",
            "username": "moshthepitt",
            "website": "",
          },
          "user": Object {
            "email": "mosh@example.com",
            "gravatar": "https://secure.gravatar.com/avatar/ae22ab897231db07205bd5d00e64cbbf?d=https%3A%2F%2Fona.io%2Fstatic%2Fimages%2Fdefault_avatar.png&s=60",
            "name": "mosh",
            "username": "moshthepitt",
          },
        }
      }
      user={
        Object {
          "api_token": "the api token",
          "city": "",
          "country": "",
          "email": "mosh@example.com",
          "gravatar": "https://secure.gravatar.com/avatar/ae22ab897231db07205bd5d00e64cbbf?d=https%3A%2F%2Fona.io%2Fstatic%2Fimages%2Fdefault_avatar.png&s=60",
          "name": "mosh",
          "organization": "",
          "require_auth": true,
          "temp_token": "the temp token",
          "twitter": "",
          "url": "https://stage-api.ona.io/api/v1/profiles/moshthepitt",
          "user": "https://stage-api.ona.io/api/v1/users/moshthepitt",
          "username": "moshthepitt",
          "website": "",
        }
      }
    >
      <div
        className="gatekeeper-cb"
      >
        <p
          className="gatekeeper-p"
        >
          Welcome back, 
          moshthepitt
          !
        </p>
      </div>
    </SuccessfulLogin>
  </BaseCallbackComponent>
</APICallback>
`;

exports[`gatekeeper/custom/APICallback renders correctly when loading 1`] = `
<APICallback
  ErrorComponent={[Function]}
  HTTP404Component={[Function]}
  LoadingComponent={[Function]}
  SuccessfulLoginComponent={[Function]}
  UnSuccessfulLoginComponent={[Function]}
  apiURL="http://example.com"
  authSuccess={null}
  authenticateActionCreator={[Function]}
  authenticated={true}
  recordResultActionCreator={[Function]}
  sessionData={
    Object {
      "authenticated": true,
      "extraData": Object {
        "api_token": "the api token",
        "city": "",
        "country": "",
        "email": "mosh@example.com",
        "gravatar": "https://secure.gravatar.com/avatar/ae22ab897231db07205bd5d00e64cbbf?d=https%3A%2F%2Fona.io%2Fstatic%2Fimages%2Fdefault_avatar.png&s=60",
        "name": "mosh",
        "oAuth2Data": Object {
          "access_token": "iLoveOov",
          "expires_in": "36000",
          "scope": "read write",
          "state": "abc",
          "token_type": "Bearer",
        },
        "organization": "",
        "require_auth": true,
        "temp_token": "the temp token",
        "twitter": "",
        "url": "https://stage-api.ona.io/api/v1/profiles/moshthepitt",
        "user": "https://stage-api.ona.io/api/v1/users/moshthepitt",
        "username": "moshthepitt",
        "website": "",
      },
      "user": Object {
        "email": "mosh@example.com",
        "gravatar": "https://secure.gravatar.com/avatar/ae22ab897231db07205bd5d00e64cbbf?d=https%3A%2F%2Fona.io%2Fstatic%2Fimages%2Fdefault_avatar.png&s=60",
        "name": "mosh",
        "username": "moshthepitt",
      },
    }
  }
  sessionUser={
    Object {
      "api_token": "the api token",
      "city": "",
      "country": "",
      "email": "mosh@example.com",
      "gravatar": "https://secure.gravatar.com/avatar/ae22ab897231db07205bd5d00e64cbbf?d=https%3A%2F%2Fona.io%2Fstatic%2Fimages%2Fdefault_avatar.png&s=60",
      "name": "mosh",
      "organization": "",
      "require_auth": true,
      "temp_token": "the temp token",
      "twitter": "",
      "url": "https://stage-api.ona.io/api/v1/profiles/moshthepitt",
      "user": "https://stage-api.ona.io/api/v1/users/moshthepitt",
      "username": "moshthepitt",
      "website": "",
    }
  }
>
  <BaseCallbackComponent
    LoadingComponent={[Function]}
    SuccessfulLoginComponent={[Function]}
    UnSuccessfulLoginComponent={[Function]}
    authSuccess={null}
    authenticated={true}
    sessionData={
      Object {
        "authenticated": true,
        "extraData": Object {
          "api_token": "the api token",
          "city": "",
          "country": "",
          "email": "mosh@example.com",
          "gravatar": "https://secure.gravatar.com/avatar/ae22ab897231db07205bd5d00e64cbbf?d=https%3A%2F%2Fona.io%2Fstatic%2Fimages%2Fdefault_avatar.png&s=60",
          "name": "mosh",
          "oAuth2Data": Object {
            "access_token": "iLoveOov",
            "expires_in": "36000",
            "scope": "read write",
            "state": "abc",
            "token_type": "Bearer",
          },
          "organization": "",
          "require_auth": true,
          "temp_token": "the temp token",
          "twitter": "",
          "url": "https://stage-api.ona.io/api/v1/profiles/moshthepitt",
          "user": "https://stage-api.ona.io/api/v1/users/moshthepitt",
          "username": "moshthepitt",
          "website": "",
        },
        "user": Object {
          "email": "mosh@example.com",
          "gravatar": "https://secure.gravatar.com/avatar/ae22ab897231db07205bd5d00e64cbbf?d=https%3A%2F%2Fona.io%2Fstatic%2Fimages%2Fdefault_avatar.png&s=60",
          "name": "mosh",
          "username": "moshthepitt",
        },
      }
    }
    sessionUser={
      Object {
        "api_token": "the api token",
        "city": "",
        "country": "",
        "email": "mosh@example.com",
        "gravatar": "https://secure.gravatar.com/avatar/ae22ab897231db07205bd5d00e64cbbf?d=https%3A%2F%2Fona.io%2Fstatic%2Fimages%2Fdefault_avatar.png&s=60",
        "name": "mosh",
        "organization": "",
        "require_auth": true,
        "temp_token": "the temp token",
        "twitter": "",
        "url": "https://stage-api.ona.io/api/v1/profiles/moshthepitt",
        "user": "https://stage-api.ona.io/api/v1/users/moshthepitt",
        "username": "moshthepitt",
        "website": "",
      }
    }
  >
    <RenderLoadingComponent>
      <div
        className="gatekeeper-cb"
      >
        <p
          className="gatekeeper-p"
        >
          Please wait...
        </p>
      </div>
    </RenderLoadingComponent>
  </BaseCallbackComponent>
</APICallback>
`;

exports[`gatekeeper/custom/APICallback renders correctly when not authenticated 1`] = `
<APICallback
  ErrorComponent={[Function]}
  HTTP404Component={[Function]}
  LoadingComponent={[Function]}
  SuccessfulLoginComponent={[Function]}
  UnSuccessfulLoginComponent={[Function]}
  apiURL="http://example.com"
  authSuccess={true}
  authenticateActionCreator={[Function]}
  authenticated={false}
  recordResultActionCreator={[Function]}
  sessionData={
    Object {
      "authenticated": true,
      "extraData": Object {
        "api_token": "the api token",
        "city": "",
        "country": "",
        "email": "mosh@example.com",
        "gravatar": "https://secure.gravatar.com/avatar/ae22ab897231db07205bd5d00e64cbbf?d=https%3A%2F%2Fona.io%2Fstatic%2Fimages%2Fdefault_avatar.png&s=60",
        "name": "mosh",
        "oAuth2Data": Object {
          "access_token": "iLoveOov",
          "expires_in": "36000",
          "scope": "read write",
          "state": "abc",
          "token_type": "Bearer",
        },
        "organization": "",
        "require_auth": true,
        "temp_token": "the temp token",
        "twitter": "",
        "url": "https://stage-api.ona.io/api/v1/profiles/moshthepitt",
        "user": "https://stage-api.ona.io/api/v1/users/moshthepitt",
        "username": "moshthepitt",
        "website": "",
      },
      "user": Object {
        "email": "mosh@example.com",
        "gravatar": "https://secure.gravatar.com/avatar/ae22ab897231db07205bd5d00e64cbbf?d=https%3A%2F%2Fona.io%2Fstatic%2Fimages%2Fdefault_avatar.png&s=60",
        "name": "mosh",
        "username": "moshthepitt",
      },
    }
  }
  sessionUser={
    Object {
      "api_token": "the api token",
      "city": "",
      "country": "",
      "email": "mosh@example.com",
      "gravatar": "https://secure.gravatar.com/avatar/ae22ab897231db07205bd5d00e64cbbf?d=https%3A%2F%2Fona.io%2Fstatic%2Fimages%2Fdefault_avatar.png&s=60",
      "name": "mosh",
      "organization": "",
      "require_auth": true,
      "temp_token": "the temp token",
      "twitter": "",
      "url": "https://stage-api.ona.io/api/v1/profiles/moshthepitt",
      "user": "https://stage-api.ona.io/api/v1/users/moshthepitt",
      "username": "moshthepitt",
      "website": "",
    }
  }
>
  <BaseCallbackComponent
    LoadingComponent={[Function]}
    SuccessfulLoginComponent={[Function]}
    UnSuccessfulLoginComponent={[Function]}
    authSuccess={true}
    authenticated={false}
    sessionData={
      Object {
        "authenticated": true,
        "extraData": Object {
          "api_token": "the api token",
          "city": "",
          "country": "",
          "email": "mosh@example.com",
          "gravatar": "https://secure.gravatar.com/avatar/ae22ab897231db07205bd5d00e64cbbf?d=https%3A%2F%2Fona.io%2Fstatic%2Fimages%2Fdefault_avatar.png&s=60",
          "name": "mosh",
          "oAuth2Data": Object {
            "access_token": "iLoveOov",
            "expires_in": "36000",
            "scope": "read write",
            "state": "abc",
            "token_type": "Bearer",
          },
          "organization": "",
          "require_auth": true,
          "temp_token": "the temp token",
          "twitter": "",
          "url": "https://stage-api.ona.io/api/v1/profiles/moshthepitt",
          "user": "https://stage-api.ona.io/api/v1/users/moshthepitt",
          "username": "moshthepitt",
          "website": "",
        },
        "user": Object {
          "email": "mosh@example.com",
          "gravatar": "https://secure.gravatar.com/avatar/ae22ab897231db07205bd5d00e64cbbf?d=https%3A%2F%2Fona.io%2Fstatic%2Fimages%2Fdefault_avatar.png&s=60",
          "name": "mosh",
          "username": "moshthepitt",
        },
      }
    }
    sessionUser={
      Object {
        "api_token": "the api token",
        "city": "",
        "country": "",
        "email": "mosh@example.com",
        "gravatar": "https://secure.gravatar.com/avatar/ae22ab897231db07205bd5d00e64cbbf?d=https%3A%2F%2Fona.io%2Fstatic%2Fimages%2Fdefault_avatar.png&s=60",
        "name": "mosh",
        "organization": "",
        "require_auth": true,
        "temp_token": "the temp token",
        "twitter": "",
        "url": "https://stage-api.ona.io/api/v1/profiles/moshthepitt",
        "user": "https://stage-api.ona.io/api/v1/users/moshthepitt",
        "username": "moshthepitt",
        "website": "",
      }
    }
  >
    <RenderErrorComponent>
      <div
        className="gatekeeper-cb"
      >
        <p
          className="gatekeeper-p"
        >
          An error occurred!
        </p>
      </div>
    </RenderErrorComponent>
  </BaseCallbackComponent>
</APICallback>
`;
